connector/src/main/java/org/jboss/as/connector/services/workmanager/StatisticsExecutorImpl.java

```diff
@@ -85,25 +85,25 @@ public void executeNonBlocking(Runnable runnable) throws RejectedExecutionExcept
     @Override
     public long getNumberOfFreeThreads() {
         if (realExecutor instanceof JBossThreadPoolExecutor) {
-            return ((JBossThreadPoolExecutor) realExecutor).getMaximumPoolSize() -
+            return (long) ((JBossThreadPoolExecutor) realExecutor).getMaximumPoolSize() -
                     ((JBossThreadPoolExecutor) realExecutor).getActiveCount();
         } else if (realExecutor instanceof ThreadPoolExecutorMBean) {
```

Comment:
you should remove all those else too

Reply:
I can provide a fix if you want. I have doubt to mix a refactoring (logic/deprecates) and a Jira adressing math. Is that OK?
Could you give me a hint why only JBossThreadPoolExecutor shall remain?

Reply:
no I meant that since every 'else if' has a return statement, the else part is superfluous. Not a big deal though